core/go/internal/sequencer/coordinator/state_machine.go

```diff
@@ -0,0 +1,491 @@
+						To: State_Observing,
+					}},
+				},
+			},
+		},
+		State_Observing: {
+			OnTransitionTo: action_StopHeartbeating,
+			Events: map[EventType]EventHandler{
+				common.Event_HeartbeatInterval: {},
+				Event_TransactionsDelegated: {
+					Transitions: []Transition{
+						{
+							To: State_Standby,
+							If: guard_Behind,
+						},
+						{
+							To: State_Elect,
+							If: guard_Not(guard_Behind),
+						},
+					},
+				},
+			},
+		},
+		State_Standby: {
+			Events: map[EventType]EventHandler{
+				common.Event_HeartbeatInterval: {},
+				Event_TransactionsDelegated:    {},
+				Event_NewBlock: {
+					Transitions: []Transition{{
+						To: State_Elect,
+						If: guard_Not(guard_Behind),
+					}},
+				},
+			},
+		},
+		State_Elect: {
+			OnTransitionTo: action_SendHandoverRequest,
+			Events: map[EventType]EventHandler{
+				common.Event_HeartbeatInterval: {},
+				Event_TransactionsDelegated:    {},
+				Event_HandoverReceived: {
+					Transitions: []Transition{{
+						To: State_Prepared,
+					}},
+				},
+			},
+		},
+		State_Prepared: {
+			Events: map[EventType]EventHandler{
+				common.Event_HeartbeatInterval: {},
+				Event_TransactionsDelegated:    {},
+				Event_TransactionConfirmed: {
+					Transitions: []Transition{{
+						To: State_Active,
+						If: guard_ActiveCoordinatorFlushComplete,
+					}},
+				},
+			},
+		},
+		State_Active: {
+			OnTransitionTo: action_SelectTransaction,
+			Events: map[EventType]EventHandler{
+				common.Event_HeartbeatInterval: {
+					Actions: []ActionRule{{
+						Action: action_SendHeartbeat,
+					}},
+				},
+				Event_TransactionsDelegated: {
+					Actions: []ActionRule{{
+						Action: action_SelectTransaction,
+						If:     guard_Not(guard_HasTransactionAssembling),
+					}},
+				},
+				Event_TransactionConfirmed: {
+					Transitions: []Transition{{
+						To: State_Idle,
+						If: guard_Not(guard_HasTransactionsInflight),
+					}},
+				},
+				Event_HandoverRequestReceived: { // MRW TODO - what if N nodes all startup in active mode simultaneously? None of them can request handover because that only happens from State_Observing
+					Transitions: []Transition{{
+						To: State_Flush,
+					}},
+				},
+			},
+		},
+		State_Flush: {
+			//TODO should we move to active if we get delegated transactions while in flush?
+			OnTransitionTo: action_StopHeartbeating,
+			Events: map[EventType]EventHandler{
+				common.Event_HeartbeatInterval: {},
+				Event_TransactionConfirmed: {
+					Transitions: []Transition{{
+						To: State_Closing,
+						If: guard_FlushComplete,
+					}},
+				},
+			},
+		},
+		State_Closing: {
+			//TODO should we move to active if we get delegated transactions while in closing?
+			OnTransitionTo: action_StopHeartbeating,
+			Events: map[EventType]EventHandler{
+				common.Event_HeartbeatInterval: {
+					Transitions: []Transition{{
+						To: State_Idle,
+						If: guard_ClosingGracePeriodExpired,
+					}},
+				},
+			},
+		},
```

Comment:
If these questions in comments aren't answered in the architecture currently (I know we've discussed them in whiteboard interactions), then probably critical to think about them in the protocol specification as well as the code.